DEFEDIT-1202 Search searches in all text files in project

ragnard-king opened the pull request:
### User facing changes

Fixes https://github.com/defold/editor2-issues/issues/1370, https://github.com/defold/editor2-issues/issues/607

- "Search in Files" will now search all text files in project,
  including dependencies, builtins and files that are not editied
  within editor.

### Technical changes

- we now cache save data on a background thread so the "Indexing..."
  step when loading a project is gone
- consider all resource-nodes connected to Project/nodes when
  searching
  - if they have save-data with content, search that
  - otherwise search the disk contents
  - ignore files that look like they don't have textual content

ragnard-king committed:
DEFEDIT-1202 Search searches in all text files in project

* User facing changes

- "Search in Files" will now search all text files in project,
  including dependencies, builtins and files that are not editied
  within editor.

* Technical changes

- we now cache save data on a background thread so the "Indexing..."
  step when loading a project is gone
- consider all resource-nodes connected to Project/nodes when
  searching
  - if they have save-data with content, search that
  - otherwise search the disk contents
  - ignore files that look like they don't have textual content

matgis-king reviewed (commented):
Reviewed! Some minor things and some questions but looks good overall, and it works well!

ragnard-king committed:
DEFEDIT-1202 Review fixes

- sort imports/requires
- rely on io/reader returning `BufferedReader`s instead of creating
  redundant ones
- search-data -> save-data
- sleep a bit in test between search result polls
